code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -347,7 +345,94 @@ private EdgeEndpoint getEdgeEndpoint(final Map<String, Object> edgeConfiguration
-		return new EdgeEndpoint(requestEnvironment, requestDomain, locationHint);
+
+		// Use null fallback value for request without custom path value
+		String customPath = DataReader.optString(requestProperties, EdgeConstants.EventDataKeys.Request.PATH, null);
+
+		return new EdgeEndpoint(requestType, requestEnvironment, requestDomain, customPath, locationHint);
+	}
+
+	/**
+	 * Extracts all the custom request properties to overwrite the default values
+	 * @param event current event for which the request properties are to be extracted
+	 * @return the map of extracted request properties and their custom values
+	 */
+	private Map<String, Object> getRequestProperties(final Event event) {
+		Map<String, Object> requestProperties = new HashMap<>();
+
+		String overwritePath = getCustomRequestPath(event);
+		if (overwritePath != null) {
+			Log.trace(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Got custom path:(%s) for event:(%s), which will overwrite the default interaction request path.",
+				overwritePath,
+				event.getUniqueIdentifier()
+			);
+			requestProperties.put(EdgeConstants.EventDataKeys.Request.PATH, overwritePath);
+		}
+		return requestProperties;
+	}
+
+	/**
+	 * Extracts network request path property to overwrite the default endpoint path value
+	 * @param event current event for which the request path property is to be extracted
+	 * @return the custom path string
+	 */
+	private String getCustomRequestPath(final Event event) {
+		String path = null;
+
+		Map<String, Object> eventData = event.getEventData();
+		try {
+			if (!MapUtils.isNullOrEmpty(eventData)) {
+				Map<String, Object> requestData = (Map<String, Object>) eventData.get(
+					EdgeConstants.EventDataKeys.Request.KEY
+				);
+				if (!MapUtils.isNullOrEmpty(requestData)) {
+					path = (String) requestData.get(EdgeConstants.EventDataKeys.Request.PATH);
+				}
+			}
+		} catch (ClassCastException e) {
+			Log.error(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Exception while getting custom path from the experience event (%s).",
+				e.getLocalizedMessage()
+			);
+		}
+
+		if (!isValidPath(path)) {
+			Log.error(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Dropping the overwrite path value: (%s), since it contains invalid characters or is empty or null.",
+				path
+			);
+			return null;
+		}
+
+		return path;
+	}
+
+	/**
+	 * Validates a given path does not contain invalid characters.
+	 * A 'path'  may only contain alphanumeric characters, forward slash, period, hyphen, underscore, or tilde, but may not contain a double forward slash.
+	 * @param path the path to validate
+	 * @return true if 'path' passes validation, false if 'path' contains invalid characters.
+	 */
+	private boolean isValidPath(final String path) {
+		if (path == null || path.isEmpty()) {
+			return false;
+		}
+
+		if (path.contains("//")) {
+			return false;
+		}
+
+		Pattern pattern = Pattern.compile(VALID_PATH_REGEX_PATTERN);
```

Comment:
Save this Pattern as an instance constant so we don't need to compile it for each request.